test/org/zaproxy/zap/authentication/UsernamePasswordAuthenticationCredentialsUnitTest.java

```diff
@@ -0,0 +1,121 @@
+    private String password = "myPassword";
+
+    @Before
+    public void setUp() {
+        this.usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+        this.notConfiguredInstance = new UsernamePasswordAuthenticationCredentials();
+    }
+
+    @Test
+    public void shouldBeConfiguredIfUsernameAndPasswordAreNotNull() throws Exception {
+        // Given/When
+        boolean isConfigured = usernamePasswordAuthenticationCredentials.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(true));
+    }
+
+    @Test
+    public void shouldNotBeConfiguredIfUsernameAndPasswordAreNull() throws Exception {
+        // Given/When
+        boolean isConfigured = notConfiguredInstance.isConfigured();
+
+        // Then
+        assertThat(isConfigured, is(false));
+    }
+
+
+    @Test
+    public void shouldEncodeMethodReturnNullPatternIfUsernameIsNull() throws Exception {
+        // Given
+        String nullPattern = "AA==";
+        username = null;
+        password = "something";
+        String stringSeparator = "|";
+        usernamePasswordAuthenticationCredentials = new UsernamePasswordAuthenticationCredentials(username, password);
+
+        // When
+        String encodedResult = usernamePasswordAuthenticationCredentials.encode(stringSeparator);
+
+        // Then
+        assertThat(encodedResult, is(nullPattern));
+    }
+
+    @Test
+    public void shouldEncodeUsernameAndPasswordWithANonFieldSeparator() throws Exception {
+        // Given
+        List<String> sampleNonFieldSeparators = Arrays.asList("-", "|", "/", "\\", "+");
+        String sampleNonFieldSeparator = sampleNonFieldSeparators.get(new Random().nextInt(sampleNonFieldSeparators.size()));
+
+        // When
+        String encodedUsernamePassword = usernamePasswordAuthenticationCredentials.encode(sampleNonFieldSeparator);
+
+        // Then
+        assertThat(encodedUsernamePassword, not(isEmptyOrNullString()));
+        assertThat(encodedUsernamePassword, allOf(containsString("bXlVc2VybmFtZQ=="),
+                                                    containsString("bXlQYXNzd29yZA=="),
+                                                    containsString("~")));
+    }
+
+    @Test
+    public void shouldApiResponseRepresentationReturnApiResponseWithValidNameAndJsonFormat() throws Exception {
+        // Given/When
+        ApiResponse apiResponse = usernamePasswordAuthenticationCredentials.getApiResponseRepresentation();
+        JSON jsonRepresentation = apiResponse.toJSON();
+
+        // Then
+        assertThat(apiResponse, notNullValue());
+        assertThat(apiResponse.getName(), equalToIgnoringCase("credentials"));
+        assertThat(jsonRepresentation.toString(), containsString("myUsername"));
+        assertThat(jsonRepresentation.toString(), containsString("myPassword"));
+        assertThat(jsonRepresentation.toString(), containsString("type"));
```

Comment:
Is it worth assigning `jsonRepresentation.toString()` to a String variable instead of converting it 3 times?